mathics/core/formatter.py

```diff
@@ -119,3 +151,386 @@ def add_conversion_fn(cls, module_fn_name=None) -> None:
+            elif text in ("\u2146", "\U0001D451"):
+                return r"\, d"
+            elif text == "\u2211":
+                return r"\sum"
+            elif text == "\u220f":
+                return r"\prod"
+            elif len(text) > 1:
+                return render(r"\text{%s}", text, in_text=True)
+            else:
+                return render("%s", text)
+
+    def get_head(self) -> Symbol:
+        return SymbolString
+
+    def get_head_name(self) -> str:
+        return "System`String"
+
+    def get_string_value(self) -> str:
+        return self.value
+
+    def to_expression(self) -> String:
+        return String(self.value)
+
+
+element_formatters = {}
+
+
+def format_element(
+    element: BaseElement, evaluation: Evaluation, form: Symbol, **kwargs
+) -> BaseElement:
+    """
+    Applies formats associated to the expression, and then calls Makeboxes
+    """
+    do_format = element_formatters.get(type(element), do_format_element)
+    expr = do_format(element, evaluation, form)
+    result = Expression(SymbolMakeBoxes, expr, form)
+    result_box = result.evaluate(evaluation)
+
+    if isinstance(result_box, BoxElement):
+        return result_box
+    elif isinstance(result_box, String):
+        return _BoxedString(result_box.value)
+    else:
+        return format_element(element, evaluation, SymbolFullForm, **kwargs)
+
+
+# do_format_*
+
+
+def do_format(
+    element: BaseElement, evaluation: Evaluation, form: Symbol
+) -> BaseElement:
+    do_format_method = element_formatters.get(type(element), do_format_element)
+    return do_format_method(element, evaluation, form)
+
+
+def do_format_element(
+    element: BaseElement, evaluation: Evaluation, form: Symbol
+) -> BaseElement:
+    """
+    Applies formats associated to the expression and removes
+    superfluous enclosing formats.
+    """
+
+    formats = format_symbols
+    evaluation.inc_recursion_depth()
+    try:
+        expr = element
+        head = element.get_head()
+        elements = element.get_elements()
+        include_form = False
+        # If the expression is enclosed by a Format
+        # takes the form from the expression and
+        # removes the format from the expression.
+        if head in formats and len(elements) == 1:
+            expr = elements[0]
+            if not (form is SymbolOutputForm and head is SymbolStandardForm):
+                form = head
+                include_form = True
+
+        # If form is Fullform, return it without changes
+        if form is SymbolFullForm:
+            if include_form:
+                expr = Expression(form, expr)
+            return expr
+        # Repeated and RepeatedNull confuse the formatter,
+        # so we need to hardlink their format rules:
+        if head is SymbolRepeated:
+            if len(elements) == 1:
+                return Expression(
+                    SymbolHoldForm,
+                    Expression(
```

Comment:
This is not a valid Expression. Use `Integer() `and `String()` to convert arguments or use `to_expression()`